tests/testconnmngr.nim

```diff
@@ -187,10 +187,45 @@ suite "Connection Manager":
 
     expect TooManyConnectionsError:
       connMngr.storeConn(conns[0])
+
+    await connMngr.close()
+
+    await allFuturesThrowing(
+      allFutures(conns.mapIt( it.close() )))
+
+  asyncTest "expect connection from peer":
+    let connMngr = ConnManager.new(maxConnsPerPeer = 0)
```

Comment:
This is even more confusing, IMHO. If we don't want to fix this bug in this PR, I would keep it as 1 and add a comment explaining the bug with a FIXME.

Reply:
Except that we really need a limit of 1 for this test to pass

Reply:
I mean, with a limit of 0, this test should fail, shouldn't it?